dhis_android_analytics/src/main/java/dhis2/org/analytics/charts/ChartsRepositoryImpl.kt

```diff
@@ -134,12 +139,34 @@ class ChartsRepositoryImpl(
                         withPeriods(dimensionPeriods)
                     } ?: this
                 }
+                .run {
+                    when (selectedOrgUnitType) {
+                        OrgUnitFilterType.ALL -> {
+                            withOrganisationUnits(
```

Comment:
Is there an specific reason of using a second `.run { }` when the 1st run is returning the same object?

Reply:
I had to ask Victor becouse my intention was using `apply{ }`. It seems this filters return a different object and apply did not worked. Victor told me to try with `run{}`